db/migrate/20150405012126_remove_approval_status_from_reservation.rb

```diff
@@ -0,0 +1,44 @@
+class RemoveApprovalStatusFromReservation < ActiveRecord::Migration
+  def change
+    Reservation.all.each do |res|
+      Reservation.transaction begin
+        if res.approval_status == 'approved'
+          res.flag(:request)
+        elsif res.approval_status == 'missed_and_emailed'
+          res.flag(:missed_email_sent)
+        end
+
+        if res.checked_in 
+          res.status = 'returned'
+          if res.checked_in > (res.due_date + 1.day)
+            res.overdue = true
+          end
+          if res.notes.include? 'Archived'
+            res.status = 'archived'
+          end
+        elsif res.checked_out
+          res.status = 'checked_out'
+          if (res.due_date + 1.day) < Time.zone.today
+            res.overdue = true
+          end
+        elsif res.start_date < Time.zone.today
+          res.status = 'missed'
+        elsif res.approval_status == 'requested'
+          res.flag(:request)
+          res.status = 'requested'
+        elsif res.approval_status == 'denied'
+          res.flag(:request)
+          res.status = 'denied'
+        elsif 'approved auto'.include? res.approval_status
+          res.status = 'reserved'
+        else
```

Comment:
do me a favor, just comment here saying that this is so to catch any unusual reservations